stl/src/StlLCMapStringW.cpp

```diff
@@ -39,10 +39,10 @@ extern "C" int __cdecl __crtLCMapStringW(_In_opt_z_ LPCWSTR const locale_name, _
         const int source_length = static_cast<int>(wcsnlen(source, source_count));
 
         // Include the null terminator if the source string is terminated within the buffer.
-        if (source_length < source_count) {
-            source_count = source_length + 1;
-        } else {
+        if (source_length > source_count) {

```

Comment:
IIUC this condition is never satisified. According to [MS CRT document](https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/strnlen-strnlen-s?view=msvc-170#return-value) and [POSIX document](https://pubs.opengroup.org/onlinepubs/9699919799/functions/wcslen.html), `wcsnlen` never returns a value greater than `source_count`.